src/i16x16_.rs

```diff
@@ -0,0 +1,504 @@
+        Self { sse0: cmp_gt_mask_i16_m128i(self.sse0, rhs.sse0),
+          sse1: cmp_gt_mask_i16_m128i(self.sse1, rhs.sse1) }
+      } else {
+        Self { arr: [
+          if self.arr[0] > rhs.arr[0] { -1 } else { 0 },
+          if self.arr[1] > rhs.arr[1] { -1 } else { 0 },
+          if self.arr[2] > rhs.arr[2] { -1 } else { 0 },
+          if self.arr[3] > rhs.arr[3] { -1 } else { 0 },
+          if self.arr[4] > rhs.arr[4] { -1 } else { 0 },
+          if self.arr[5] > rhs.arr[5] { -1 } else { 0 },
+          if self.arr[6] > rhs.arr[6] { -1 } else { 0 },
+          if self.arr[7] > rhs.arr[7] { -1 } else { 0 },
+          if self.arr[8] > rhs.arr[8] { -1 } else { 0 },
+          if self.arr[9] > rhs.arr[9] { -1 } else { 0 },
+          if self.arr[10] > rhs.arr[10] { -1 } else { 0 },
+          if self.arr[11] > rhs.arr[11] { -1 } else { 0 },
+          if self.arr[12] > rhs.arr[12] { -1 } else { 0 },
+          if self.arr[13] > rhs.arr[13] { -1 } else { 0 },
+          if self.arr[14] > rhs.arr[14] { -1 } else { 0 },
+          if self.arr[15] > rhs.arr[15] { -1 } else { 0 },
+        ]}
+      }
+    }
+  }
+}
+
+impl CmpLt for i16x16 {
+  type Output = Self;
+  #[inline]
+  #[must_use]
+  fn cmp_lt(self, rhs: Self) -> Self::Output {
+    pick! {
+      if #[cfg(target_feature="avx2")] {
+        //TODO: what about == ?
+        Self { avx2: !cmp_gt_mask_i16_m256i(self.avx2, rhs.avx2) }
+      } else if #[cfg(target_feature="sse2")] {
+        Self { sse0: cmp_lt_mask_i16_m128i(self.sse0, rhs.sse0),
+          sse1: cmp_lt_mask_i16_m128i(self.sse1, rhs.sse1) }
+      } else {
+        Self { arr: [
+          if self.arr[0] < rhs.arr[0] { -1 } else { 0 },
+          if self.arr[1] < rhs.arr[1] { -1 } else { 0 },
+          if self.arr[2] < rhs.arr[2] { -1 } else { 0 },
+          if self.arr[3] < rhs.arr[3] { -1 } else { 0 },
+          if self.arr[4] < rhs.arr[4] { -1 } else { 0 },
+          if self.arr[5] < rhs.arr[5] { -1 } else { 0 },
+          if self.arr[6] < rhs.arr[6] { -1 } else { 0 },
+          if self.arr[7] < rhs.arr[7] { -1 } else { 0 },
+          if self.arr[8] < rhs.arr[8] { -1 } else { 0 },
+          if self.arr[9] < rhs.arr[9] { -1 } else { 0 },
+          if self.arr[10] < rhs.arr[10] { -1 } else { 0 },
+          if self.arr[11] < rhs.arr[11] { -1 } else { 0 },
+          if self.arr[12] < rhs.arr[12] { -1 } else { 0 },
+          if self.arr[13] < rhs.arr[13] { -1 } else { 0 },
+          if self.arr[14] < rhs.arr[14] { -1 } else { 0 },
+          if self.arr[15] < rhs.arr[15] { -1 } else { 0 },
+        ]}
+      }
+    }
+  }
+}
+
+impl i16x16 {
+  #[inline]
+  #[must_use]
+  pub fn blend(self, t: Self, f: Self) -> Self {
+    pick! {
+      if #[cfg(target_feature="avx2")] {
+        Self { avx2: blend_varying_i8_m256i(f.avx2, t.avx2, self.avx2) }
+      } else if #[cfg(target_feature="sse4.1")] {
+        Self { sse: blend_varying_i8_m128i(f.sse, t.sse, self.sse) }
```

Comment:
Is this right that blend should be i8?

Reply:
an i8 blend should be fine if the mask is an i16 mask. each two i8 lanes of the mask will be identical, so they'll use the identical source for the output, so it ends up working like an i16 blend.